### DISCUSSION TITLE
Does this support model statistics or any plans to add.

### DISCUSSION THREAD

**Discussion by gurvinder-dhillon** [gmaxlev/pict-node#7] [Q&A]
I don't see anywhere in documentation about model statistics, so wondering if is available or needs to be added.

**Comment by gmaxlev** [comment]
@gurvinder-dhillon Unfortunately, this feature is currently not available. Initially I wanted to add it, but later changed my mind. If you really need it, I can add it.

pict provides API to get statistics separately. Therefore, I imagine getting statistics as a separate function call. For example:

```js
const statistics = pict.statistics({ model });
```
```js
const statistics = strings.statistics({ model });
```
```js
const statistics = native.statistics({ model });
```

If you have any ideas, I can consider them

**Comment by gurvinder-dhillon** [comment]
Hi @gmaxlev Thanks for your reply.
It would be great if "strings" can support this as an option.
`strings( { model, constraints }, { statistics: true, order: 2, caseSensitive: false })`

Reason for picking up strings option is because with constraints the number of test cases would change and so does the statistics.

And i think that should not be a big lift considering "/s" is supported as command line option.
https://github.com/Microsoft/pict/blob/main/doc/pict.md#usage

Again thanks for building this.

**Comment by gmaxlev** [comment]
The capability to retrieve model statistics is now available. The `pict`, `storing`, and `native` functions now include `stats` method that returns statistics, with the same parameters.

I implemented this as a separate function to avoid overloading the existing function and complicating the types.

You can find usage examples in the [documentation](https://github.com/gmaxlev/pict-node#statistics).

@gurvinder-dhillon thank you very much for the feature request, if you have any comments or suggestions, I would be happy to consider them.

**Reply by gurvinder-dhillon** [reply]
Thank you so much. I appreciate you resolving this quickly.
